pkg/controller/jobframework/reconciler.go

```diff
@@ -509,6 +519,21 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 	return true
 }
 
+func (r *JobReconciler) updateWorkloadToMatchJob(ctx context.Context, job GenericJob, object client.Object, wl *kueue.Workload) (*kueue.Workload, error) {
+	newWl, err := r.constructWorkload(ctx, job, object)
+	if err != nil {
+		return nil, fmt.Errorf("can't construct workload for update: %w", err)
+	}
+	wl.Spec = newWl.Spec
+	if err = r.client.Update(ctx, wl); err != nil {
+		return nil, fmt.Errorf("updating existed workload: %w", err)
+	}
+
+	r.record.Eventf(object, corev1.EventTypeNormal, "UpdatedWorkload",
+		"Updated not matching Workload for suspended job: %v", wl)
```

Comment:
More of a question: are we sure the job is suspended at this point? For example, i'm thinking if this code is executed when the job is running, but a user manually modified the admitted workload.

Reply:
This function is called if the job is suspended and the workload is not admitted. Or do you mean that that state of the job was changed during the Reconcile() execution? I don't think we can do anything about it.

Reply:
> This function is called if the job is suspended and the workload is not admitted.

I looked in the code a bit, but could not see any `if` that guarantees the `ensureOneWorkload` function is called only for suspended jobs.  Actually the comment `// If there's no workload exists and job is unsuspended, we'll stop it immediately.` suggests under some circumstances the job might be unsuspended. Please double check. 

If there are some scenarios in which the job is unsuspeded, maybe we should just modify the message as `"Updated not matching Workload for job: %v`.

Reply:
Right, the job which is passed to ensureOneWorkload could be unsuspended and then we may suspend it (line 427),  however for updating we check the status of job earlier (line 421).

If the job is unsuspended and workload is modified we unsuspend the job and recreate the workload. My understanding of the issue is that we want to update workloads only for suspended job.

Reply:
Oh, I see, you actually have this check inside the function `job.IsSuspended() && !workload.IsAdmitted(toDelete[0])` which makes sure the updating is only for suspended jobs.